### DISCUSSION TITLE
More information around QueryCacheNotifyEvent

### DISCUSSION THREAD

**Discussion by romellem** [TanStack/query#4635] [Q&A]
[`queryCache.subscribe(callback)`](https://tanstack.com/query/v4/docs/reference/QueryCache#querycachesubscribe) calls the `callback()` with an `event: QueryCacheNotifyEvent` argument. Here are the possible `types` that `event` might be:

https://github.com/TanStack/query/blob/3c9df2028c447f19afd748ac55769143e62d3664/packages/query-core/src/queryCache.ts#L22-L59

I'm curious about the differences between these types, and in particular what we expect to _change_ when the events are fired? Is it expected that `updated` will have its `query.state.data` changed? Or will that only happen when `observerResultsUpdated` is fired?

Basically just looking for more documentation around these event types, and what kinds of differences are expected on an underlying `query` object when one of these types are fired?

- `added`
- `removed`
- `updated`
- `observerAdded`
- `observerRemoved`
- `observerResultsUpdated`
- `observerOptionsUpdated`

**Comment by TkDodo** [comment]
`added` and `removed` events should be self explanatory, as well as `observerAdded` and `observerRemoved`.

`updated` means a query was updated. Yes, in `query` you can expect the new state, something has changed in here.

`observerOptionsUpdated` - the options of an observer have been updated. You can expect this to be fired when there is a `useQuery` and the options that are passed to it change.

`observerResultsUpdated` - when the result of an observer has been updated 😅. Not sure what much else to say here, an observer could theoretically have the same result even though the query changed, because it is subscribed to a sub-set of the query via `select`.
